public/app/features/dashboard/state/DashboardModel.ts

```diff
@@ -356,6 +359,8 @@ export class DashboardModel {
   }
 
   initEditPanel(sourcePanel: PanelModel): PanelModel {
+    this.previousAutoRefresh = this.refresh;
+    getTimeSrv().setAutoRefresh('');
```

Comment:
nit: maybe this logic belongs to `TimeSrv`,  we could have an `pauseAutoRefresh` and `resumeAutoRefresh` function that handles this logic. But at the same time it feels strange that TimeSrv should mutate dashboard state directly.

Reply:
> nit: maybe this logic belongs to `TimeSrv`, we could have an `pauseAutoRefresh` and `resumeAutoRefresh` function that handles this logic. But at the same time it feels strange that TimeSrv should mutate dashboard state directly. (But it already does)

Reply:
I agree with you,  this logic belongs to the `TimeSrv`, I  did a small refactor in the code https://github.com/grafana/grafana/pull/34128/commits/e553524da6f8cbaea5c20c12e8f270ccbade62b8 and now we are controlling the value of the `previousAutoRefresh` in the `TimeSrv`

Reply:
Beautiful 🎉